alphastats/DataSet.py

```diff
@@ -126,8 +137,16 @@ def preprocess(
 
     def reset_preprocessing(self):
         """Reset all preprocessing steps"""
-        self.create_matrix()
-        # TODO fix bug: metadata is not reset here
+        self._create_matrix()
+        self.preprocessing_info = Preprocess.init_preprocessing_info(
+            num_samples=self.mat.shape[0],
+            num_protein_groups=self.mat.shape[1],
+            intensity_column=self.intensity_column,
+            filter_columns=self.filter_columns,
+        )
+
+        self.preprocessed = False
+        # TODO fix bug: metadata is not reset/reloaded here
```

Comment:
I think this would fit well in this branch. If you already addressed this ona nother one, just ignore the comment.

Reply:
this is not straightforward, so I'd leave it to a later PR